frontend/src/app/components/EmailDialog.js

```diff
@@ -35,7 +35,9 @@ export default class EmailDialog extends React.Component {
   }
 
   componentDidMount() {
-    this.modalContainer.focus();
+    if (this.modalContainer !== undefined) {
```

Comment:
Noticed this change in a bunch of spots - did we end up with a race condition here or something? This should always end up [with a reference](https://github.com/fzzzy/testpilot/blob/1910271403f75fca1a52cc4664b2ab6f239908e5/frontend/src/app/components/EmailDialog.js#L48) when the component has mounted

Reply:
I, too, would like to know why this happens, but I couldn't figure it out, so this change was just making it work without really knowing why. Interestingly, @kgeorgiou seems to have made similar changes in https://github.com/mozilla/testpilot/pull/3082/files. I thought the behavior change was needed because of the update to enzyme but if @kgeorgiou needed to make the same change without updating react and enzyme, ... ?

Reply:
I recently found out that for refs to work in Enzyme the components have to be rendered with `mount`, instead of `shallow` ([source](https://github.com/airbnb/enzyme/issues/316#issuecomment-208521009)). 
So by using `subject = mount(<SomeComponent {...{ onDismiss, sendToGA, getWindowLocation }} />);` will ensure that refs are evaluated as expected.